compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -924,9 +924,18 @@ object Parsers {
           val tparams = typeParamClause(ParamOwner.TypeParam)
           if (in.token == TLARROW)
             atSpan(start, in.skipToken())(LambdaTypeTree(tparams, toplevelTyp()))
-          else if (isIdent && in.name.toString == "->")
-            atSpan(start, in.skipToken())(PolyFunction(tparams, toplevelTyp()))
-          else { accept(TLARROW); typ() }
+          else if (in.token == ARROW) {
```

Comment:
We should update the syntax in the doc comment to account for this, and do the same in file `syntax.md`. Generally, we try parser code, parser comments and syntax file in sync.

Reply:
Done.